### ISSUE TITLE
Cannot run with Tomcat 7 Maven's plugin: template not found

### ISSUE THREAD

**Issue by jindrichmynarz** [opendatacz/pc-filing-app#47]
When run with Tomcat 7 Maven plugin (`mvn tomcat7:run`) the application fails with the following error:

```
HTTP Status 500 - Template '/cz/opendata/tenderstats/sparql/prefixes.mustache' not found

SEVERE: Servlet.service() for servlet [SystemManager] in context with path [] threw exception
com.github.mustachejava.MustacheException: Template '/cz/opendata/tenderstats/sparql/prefixes.mustache' not found
    at com.github.mustachejava.DefaultMustacheFactory.getReader(DefaultMustacheFactory.java:137)
    at com.github.mustachejava.MustacheParser.compile(MustacheParser.java:27)
    at com.github.mustachejava.DefaultMustacheFactory.compilePartial(DefaultMustacheFactory.java:276)
    at com.github.mustachejava.codes.PartialCode.init(PartialCode.java:70)
    at com.github.mustachejava.codes.DefaultCode.init(DefaultCode.java:118)
    at com.github.mustachejava.codes.DefaultMustache.init(DefaultMustache.java:64)
    at com.github.mustachejava.DefaultMustacheFactory.compile(DefaultMustacheFactory.java:224)
    at com.github.mustachejava.DefaultMustacheFactory.compile(DefaultMustacheFactory.java:218)
    at cz.opendata.tenderstats.Mustache.getByRelativePath(Mustache.java:28)
    at cz.opendata.tenderstats.Mustache.getBySparqlPath(Mustache.java:45)
    at cz.opendata.tenderstats.UserContext.fetchUserByEmailAndRole(UserContext.java:83)
    at cz.opendata.tenderstats.SystemManager.checkLogin(SystemManager.java:47)
    at cz.opendata.tenderstats.SystemManager.doGetPost(SystemManager.java:191)
    at cz.opendata.tenderstats.AbstractComponent.doPost(AbstractComponent.java:167)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:641)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:722)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:305)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
    at cz.opendata.tenderstats.FlashMessage.doFilter(FlashMessage.java:40)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:243)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
    at org.apache.catalina.filters.SetCharacterEncodingFilter.doFilter(SetCharacterEncodingFilter.java:108)
    at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:243)
    at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:210)
    at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:222)
    at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:123)
    at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:472)
    at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:168)
    at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:99)
    at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:929)
    at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:118)
    at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:407)
    at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1002)
    at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:585)
    at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:312)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:724)
```

**Comment by jindrichmynarz** [comment]
This seems not to be a problem with `mvn tomcat7:run-war`. However, this requires to restart the application for every change. Is it possible to [configure `mvn tomcat7:run`](http://tomcat.apache.org/maven-plugin-2.1/tomcat7-maven-plugin/run-mojo.html), so that is sets resource path properly (so that `cz.opendata.tenderstats.Mustache.getByRelativePath` doesn't throw an error)?

**Comment by propi** [comment]
Ok, I added the newest maven tomcat7 plugin to the pom.xml with the basic configuration.
The application is runnable by: mvn tomcat7:run-war

**Comment by jindrichmynarz** [comment]
As per my previous comment: the application can be run by `mvn tomcat7:run-war`, but when it's run by `mvn tomcat7:run`, then the `com.github.mustachejava.MustacheException: Template '/cz/opendata/tenderstats/sparql/prefixes.mustache' not found` is thrown. Using `mvn tomcat7:run-war` is possible, but it requires to restart the application for every change, whereas with `mvn tomcat7:run` it is automatically recompiled for every change. Is it possible to configure resource path (class path) so that `mvn tomcat7:run` doesn't throw a not found error for Mustache templates?

**Comment by jindrichmynarz** [comment]
Closed by ba2d2f1dfcbb68cc3510afc7da1e911a3430189a.
